File: app/src/main/java/com/ganesh/androidfundamentals/workmanager/backgroundwork/NetworkChangeReceiver.java
package com.ganesh.androidfundamentals.workmanager.backgroundwork;

import android.annotation.SuppressLint;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.util.Log;

public class NetworkChangeReceiver extends BroadcastReceiver {

    @SuppressLint("LogNotTimber")
    @Override
    public void onReceive(final Context context, final Intent intent) {

        final ConnectivityManager connMgr = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

        final android.net.NetworkInfo wifi = connMgr.getNetworkInfo(ConnectivityManager.TYPE_WIFI);

        final android.net.NetworkInfo mobile = connMgr.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);

        if (wifi.isAvailable() || mobile.isAvailable()) {
            Log.d("Ganesh", "Flag No 1");
        }
    }
}

File: app/src/main/java/com/ganesh/androidfundamentals/workmanager/backgroundwork/NetworkChangeReceiver.kt
package com.ganesh.androidfundamentals.workmanager.backgroundwork

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class NetworkChangeReceiver : BroadcastReceiver() {
    @SuppressLint("LogNotTimber")
    override fun onReceive(context: Context, intent: Intent) {
        Log.d("Ganesh", "onReceive NetworkChangeReceiver")
    }
}